securedrop/journalist_app/admin.py

```diff
@@ -35,24 +35,14 @@ def manage_config():
             f = form.logo.data
             static_filepath = os.path.join(config.SECUREDROP_ROOT,
                                            "static/i/logo.png")
-            temp_static_filepath = os.path.join(config.SECUREDROP_ROOT,
-                                                "static/i/temp_logo.png")
-            with open(static_filepath) as img_file:
-                current_img = img_file.read()
-            f.save(static_filepath)
-
             try:
-                with Image.open(static_filepath) as im:
+                with Image.open(f) as im:
```

Comment:
I'm not sure why you're copying the image and than saving the copy. The original is still being overwritten so we can still end up in a place where we have a broken image.

Reply:
I was trying to work with a copy of image instead of modifying it in place. That is unnecessary though, I'll remove it.